stl/inc/expected

```diff
@@ -0,0 +1,1035 @@
+        _THROW(bad_expected_access{_STD move(_Unexpected)});
+    }
+#endif // !__cpp_explicit_this_parameter
+
+    union {
+        _Ty _Value;
+        _Err _Unexpected;
+    };
+    bool _Has_value;
+};
+
+template <class _Ty, class _Err>
+    requires is_void_v<_Ty>
+class expected<_Ty, _Err> {
+public:
+    using value_type      = _Ty;
+    using error_type      = _Err;
+    using unexpected_type = unexpected<_Err>;
+
+    template <class _Uty>
+    using rebind = expected<_Uty, error_type>;
+
+    // [expected.void.ctor]
+    constexpr expected() noexcept : _Has_value(true) {}
+
+    // clang-format off
+    constexpr expected(const expected& _Other) noexcept(is_nothrow_copy_constructible_v<_Err>) // strengthened
+        requires (!is_trivially_copy_constructible_v<_Err>)
+        : _Has_value(_Other._Has_value) {
+        if (!_Has_value) {
+            _STD construct_at(_STD addressof(_Unexpected), _Other._Unexpected);
+        }
+    }
+
+    expected(const expected& ) requires is_trivially_copy_constructible_v<_Err> = default;
+
+    constexpr expected(expected&& _Other) noexcept(is_nothrow_move_constructible_v<_Err>) // strengthened
+        requires (!is_trivially_move_constructible_v<_Err> && is_move_constructible_v<_Err>)
+        : _Has_value(_Other._Has_value) {
+        if (!_Has_value) {
+            _STD construct_at(_STD addressof(_Unexpected), _STD move(_Other._Unexpected));
+        }
+    }
+
+    expected(expected&&) requires is_trivially_move_constructible_v<_Err> = default;
+    // clang-format on
+
+    template <class _Uty, class _UErr>
+    static constexpr bool _Is_not_convertible = !is_constructible_v<unexpected<_Err>, expected<_Uty, _UErr>&> //
+                                             && !is_constructible_v<unexpected<_Err>, expected<_Uty, _UErr>> //
+                                             && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>&> //
+                                             && !is_constructible_v<unexpected<_Err>, const expected<_Uty, _UErr>>;
+
+    template <class _Uty, class _UErr>
+        requires is_void_v<_Uty> && is_constructible_v<_Err, const _UErr&> && _Is_not_convertible<_Uty, _UErr>
+    constexpr explicit(!is_convertible_v<const _UErr&, _Err>) expected(const expected<_Uty, _UErr>& _Other) noexcept(
+        is_nothrow_constructible_v<_Err, const _UErr&>) // strengthened
+        : _Has_value(_Other._Has_value) {
+        if (!_Has_value) {
+            _Unexpected = _Other._Unexpected;
+        }
+    }
+
+    template <class _Uty, class _UErr>
+        requires is_void_v<_Uty> && is_constructible_v<_Err, _UErr> && _Is_not_convertible<_Uty, _UErr>
+    constexpr explicit(!is_convertible_v<_UErr, _Err>)
+        expected(expected<_Uty, _UErr>&& _Other) noexcept(is_nothrow_constructible_v<_Err, _UErr>) // strengthened
+        : _Has_value(_Other._Has_value) {
+        if (!_Has_value) {
+            _Unexpected = _STD forward<_UErr>(_Other._Unexpected);

```

Comment:
Note to reviewers: I forgot to adopt this place in the last round but did not want to restart a test round just for this. Will be fixed with the next batch of reviews